File: frontend/src/app/shared/utils/address.utils.ts
// Formats a full address string to display only street number, street name, and city.

export function formatAddress(fullAddress: string): string {
  // Split the address by commas
  const parts = fullAddress.split(',').map(p => p.trim());

  if (parts.length < 3) {
    return fullAddress; // Return as-is if format is unexpected
  }

  // Extract: number, street name, and city
  // Typically: "4, Cika Stevina, ..., Novi Sad, ..."
  const number = parts[0];

  // Check if number only contains digits
  const isValidNumber = /^\d+$/.test(number);
  if (!isValidNumber) {
    const street = parts[0];
  }

  const street = parts[1];
  const city = parts[parts.length - 6];

  return `${number}, ${street}, ${city}`;
}

File: mobile/app/src/main/java/com/example/ridenow/util/AddressUtils.java
package com.example.ridenow.util;

public class AddressUtils {

    /**
     * Formats a full address to a shorter format showing only number, street, and city
     * @param fullAddress The full address string
     * @return Formatted address in format "number, street, city"
     */
    public static String formatAddress(String fullAddress) {
        if (fullAddress == null || fullAddress.trim().isEmpty()) {
            return fullAddress;
        }

        // Split the address by commas
        String[] parts = fullAddress.split(",");
        for (int i = 0; i < parts.length; i++) {
            parts[i] = parts[i].trim();
        }

        if (parts.length < 3) {
            return fullAddress; // Return as-is if format is unexpected
        }

        // Extract: number, street name, and city
        // Typically: "4, Cika Stevina, ..., Novi Sad, ..."
        String number = parts[0];

        // Check if number only contains digits
        boolean isValidNumber = number.matches("^\\d+$");
        if (!isValidNumber) {
            // If first part is not a number, treat it as street name
            String street = parts[0];
            String city = parts.length >= 6 ? parts[parts.length - 6] : parts[parts.length - 1];
            return street + ", " + city;
        }

        String street = parts[1];
        String city = parts.length >= 6 ? parts[parts.length - 6] : parts[parts.length - 1];

        return number + ", " + street + ", " + city;
    }
}
